src/main/java/org/springframework/integration/aws/inbound/kinesis/KclMessageDrivenChannelAdapter.java

```diff
@@ -364,26 +365,34 @@ public Duration waitPeriodToDeleteFormerStreams() {
 
 				};
 
-		private final List<StreamConfig> streamConfigs =
-				Arrays.stream(KclMessageDrivenChannelAdapter.this.streams)
-						.map(streamName ->
-								new StreamConfig(StreamIdentifier.singleStreamInstance(streamName),
-										KclMessageDrivenChannelAdapter.this.streamInitialSequence))
-						.toList();
+		private List<StreamConfig> streamConfigs = new ArrayList<>();
 
 		StreamsTracker() {
 		}
 
+		private StreamConfig buildStreamConfig(String streamName) {
+			logger.info(LogMessage.format("Initializing stream config for Stream: [%s] for Multi Stream Tracker", streamName));
+
+			DescribeStreamSummaryResponse streamSummaryResponse = KclMessageDrivenChannelAdapter.this.kinesisClient.describeStreamSummary(DescribeStreamSummaryRequest.builder().streamName(streamName).build()).join();
+			return new StreamConfig(StreamIdentifier.multiStreamInstance(Arn.fromString(streamSummaryResponse.streamDescriptionSummary().streamARN()), streamSummaryResponse.streamDescriptionSummary().streamCreationTimestamp().toEpochMilli()),
+					KclMessageDrivenChannelAdapter.this.streamInitialSequence);
+		}
+
 		@Override
 		public List<StreamConfig> streamConfigList() {
+			if (this.streamConfigs.size() == 0) {
+				logger.info("Lazy loading the Stream Configs for Multi stream Tracker");
+				// Lazy loading the Stream Configs, only during inquiry.
+				Arrays.stream(KclMessageDrivenChannelAdapter.this.streams).forEach(streamName -> this.streamConfigs.add(buildStreamConfig(streamName)));
```

Comment:
Looking into this one more time, I think we can leverage `reactor.core.publisher.Flux` API with all of its power of mapping, parallel processing and, of course, `cache()` in the end.
Let me know if you can do that or I'll address it on merge.

Reply:
I Wont be able to do it immediately, would require a research and can take a good amount of time. Please feel free or we can take that as a separate PR.

Reply:
No problem!
I'll push that as a separate commit and share with you here!